### PR TITLE
Support static route commands

### PR DESCRIPTION
**Pull request by anas-shami**
Support static route commands:
- ip route
- ip route vrf
- ip routing
- ip routing vrf
- ip map-hostname
- ip host hostname
- ip mroute
- ip mroute vrf
- ip multicast-routing
- ip multicast-routing vrf

### CODE REVIEW

**Comment by anas-shami** [comment]
@samerd please review

File: lib/ansible/modules/network/onyx/onyx_static_route.py [hunk 1]
```diff
@@ -379,35 +381,36 @@ def generate_commands(self):
                 current_route = current_config.get(network_prefix)
             required_route = {
                 'network_prefix': network_prefix,
-                'netmask': required_config.get('netmask'),
-                'nexthop': required_config.get('nexthop')
+                'netmask': netmask,
+                'nexthop': nexthop
             }
+            required_str = network_prefix + ' ' + netmask + ' ' + nexthop
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_static_route.py]:
```python
required_str = ' '.join((network_prefix, netmask, nexthop))
```

File: lib/ansible/modules/network/onyx/onyx_static_route.py [hunk 2]
```diff
@@ -379,35 +381,36 @@ def generate_commands(self):
...
 
             ''' [route_distance, mroute_preference] one of them will be setted or None '''
             if route_distance is not None:
                 required_route['route_distance'] = route_distance
+                required_str = required_str + ' ' + str(route_distance)
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_static_route.py]: use join or print formatting, much efficient than addition:
```python
required_str = '%s %s' % (required_str, route_distance)
```

File: lib/ansible/modules/network/onyx/onyx_static_route.py [hunk 3]
```diff
@@ -379,35 +381,36 @@ def generate_commands(self):
...
             elif mroute_preference is not None:
                 required_route['mroute_preference'] = mroute_preference
-            required_str_list = list(map(str, required_route.values()))  # cast to string to able to join them
+                required_str = required_str + ' ' + str(mroute_preference)
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_static_route.py]: see above

File: lib/ansible/modules/network/onyx/onyx_static_route.py [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+            - Add a domain name to use when resolving hostnames
+        type: str
+    remove_domain:
+        description:
+            - Remove domain name
+        type: str
+        default: False
+    mroute_preference:
+        description:
+            - Assign multicast route prefernce (1-255)
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_static_route.py:87]: add validation in the code

File: lib/ansible/modules/network/onyx/onyx_static_route.py [hunk 5]
```diff
@@ -0,0 +1,461 @@
...
+        type: int
+    route_distance:
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_static_route.py:89]: add distance range 1-255 and add validation in code

**Review by samerd — CHANGES_REQUESTED** [review]
Thanks @anas-shami 
please fix according to the comments

**Comment by anas-shami** [comment]
@samerd  could you please check ?

**Review by samerd — APPROVED** [review]
Thanks @anas-shami 
looks great

**Comment by Akasurde** [comment]
Hi @anas-shami, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/mellanox.onyx<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**